build.gradle

```diff
@@ -39,6 +39,7 @@ dependencies {
 		'org.junit.jupiter:junit-jupiter-engine:5.1.0'
 	)
 
+	compileOnly 'org.checkerframework:checker-qual:2.11.0'
```

Comment:
I would remove this, as we dont use it anywhere else currenly.

Reply:
Is compileOnly inherited to other projects? I don't think so.

Reply:
Dont think so, but it would show in the source.

Reply:
https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_configurations_graph

This checker framework anno will not be shared with downstream libraries (e.g. loom, loader) and is fine to be present. `compileOnly` is exposed to neither `apiElements` (compile deps of downstream programs) or `runtimeElements` (actual run deps of downstream programs)

Reply:
I dont think we really want it in the source as mc is using JSR (we dont have that dep provided). Using JSR here like this might be a better solution, but doing what all the other projects do and not having anything like this is the best option imo.

Reply:
Again since it is not exposed to api/runtime elements, the lib will not be added to dependencies of other projects and will just appear as red annotations. I cannot use jsr because jsr does not support type annotations (e.g. indicating nullable values in maps)